src/store.js

```diff
@@ -1,14 +1,14 @@
 import { createStore, applyMiddleware, compose } from 'redux';
 import thunk from 'redux-thunk';
 import rootReducer from './redux/reducers';
-import { reduxFirestore } from 'redux-firestore';
-import { reactReduxFirebase } from 'react-redux-firebase';
+import { reduxFirestore, getFirestore } from 'redux-firestore';
+import { reactReduxFirebase, getFirebase } from 'react-redux-firebase';
 import fbConfig from './config/fbConfig';
 
 const store = createStore(
   rootReducer,
   compose(
-    applyMiddleware(thunk),
+    applyMiddleware(thunk.withExtraArgument({ getFirebase, getFirestore })),
```

Comment:
What is this middleware doing?

Reply:
it allows us to pass an extra argument to our actions creators. which in that case is an object with two properties: getFirebase and getFirestore.

Reply:
and we do that to get access to the firebase api. essentially its just an extra package for redux to work 'better' with firebase